backends/optimized/ceed-opt.c

```diff
@@ -20,10 +20,8 @@
 
 static int CeedDestroy_Opt(Ceed ceed) {
   int ierr;
-  Ceed_Opt *impl = ceed->data;
-  Ceed ceedref = impl->ceedref;
-  ierr = CeedFree(&ceedref); CeedChk(ierr);
-  ierr = CeedFree(&impl); CeedChk(ierr);
+  Ceed ceedref = ceed->delegate;
```

Comment:
Since the top-level interface sees the delegate, I think it can handle destruction.

Reply:
Good point